TLM/TLM/Util/ModsCompatibilityChecker.cs

```diff
@@ -1,82 +1,159 @@
+            {
+                IncompatibleModsPanel panel = UIView.GetAView().AddUIComponent(typeof(IncompatibleModsPanel)) as IncompatibleModsPanel;
+                panel.IncompatibleMods = detected;
+                panel.Initialize();
+                UIView.PushModal(panel);
+                UIView.SetFocus(panel);
             }
+        }
 
-            if (incompatibleMods.Count > 0) {
-                Log.Warning("Incompatible mods detected! Count: " + incompatibleMods.Count);
+        /// <summary>
+        /// Iterates installed mods looking for known incompatibilities.
+        /// </summary>
+        /// 
+        /// <returns>A list of detected incompatible mods.</returns>
+        /// 
+        /// <exception cref="ArgumentException">Invalid folder path (contains invalid characters, is empty, or contains only white spaces).</exception>
+        /// <exception cref="PathTooLongException">Path is too long (longer than the system-defined maximum length).</exception>
+        public Dictionary<PluginInfo, string> ScanForIncompatibleMods()
+        {
+            Log.Info("Scanning for incompatible mods");
 
-                if (State.GlobalConfig.Instance.Main.ScanForKnownIncompatibleModsAtStartup) {
-                    IncompatibleModsPanel panel = UIView.GetAView().AddUIComponent(typeof(IncompatibleModsPanel)) as IncompatibleModsPanel;
-                    panel.IncompatibleMods = incompatibleMods;
-                    panel.Initialize();
-                    UIView.PushModal(panel);
-                    UIView.SetFocus(panel);
+            // list of installed incompatible mods
+            Dictionary<PluginInfo, string> results = new Dictionary<PluginInfo, string>();
+
+            // only check enabled mods?
+            bool filterToEnabled = State.GlobalConfig.Instance.Main.IgnoreDisabledMods;
+
+            // iterate plugins
+            foreach (PluginInfo mod in Singleton<PluginManager>.instance.GetPluginsInfo())
+            {
+                if (!mod.isBuiltin && !mod.isCameraScript && (!filterToEnabled || mod.isEnabled))
+                {
+                    string modName = GetModName(mod);
+
+                    if (incompatibleMods.ContainsKey(mod.publishedFileID.AsUInt64))
+                    {
+                        Log.Info($"Incompatible mod: {mod.publishedFileID.AsUInt64} - {modName}");
+                        results.Add(mod, modName);
+                    }
+#if !DEBUG
+                    // Workshop TM:PE builds treat local builds as incompatible
+                    else if (mod.publishedFileID.AsUInt64 == LOCAL_MOD && (modName.Contains("TM:PE") || modName.Contains("Traffic Manager")))
+                    {
+                        Log.Info($"Local TM:PE detected: '{modName}' in '{mod.modPath}'");
+                        string folder = Path.GetDirectoryName(mod.modPath).Split(Path.DirectorySeparatorChar).Last();
```

Comment:
This is giving wrong folder from the path. Should be the name of the folder, eg. `TrafficManager` but instead it's returning the parent folder, `Mods`. Any ideas?